aiofreepybox/api/home.py

```diff
@@ -0,0 +1,266 @@
+        'node': [int],
+        'role': int
+    }
+
+    node_rule_configuration_data_schema = {
+        'roles': [node_rule_role_schema]
+    }
+
+    sms_number_data_schema = {
+        'description': 'Mon numero',
+        'phoneNumber': '',
+        'smsEnabled': bool(True),
+        'voicemailEnabled': bool(True)
+    }
+
+    sms_validation_data_schema = {
+        'applicationHash': ''
+    }
+
+    sms_number_validation_data_schema = {
+        'validationCode': ''
+    }
+
+    next_pairing_step_payload_schema = {
+        'session': int,
+        'pageid': int,
+        'fields': [object]
+    }
+
+    start_pairing_step_payload_schema = {
+        'nfc': bool(True),
+        'qrcode': bool(),
+        'type': ''
+    }
+
+    stop_pairing_step_payload_schema = {
+        'session': int,
+    }
+
+    async def del_home_adapter(self, home_adapter_id):
+        '''
+        Delete home adapter
+        '''
+        return await self._access.delete(f'home/adapters/{home_adapter_id}')
+
+    async def get_home_adapter(self, home_adapter_id):
+        '''
+        Retrieve a registered home adapter
+        '''
+        return await self._access.get(f'home/adapters/{home_adapter_id}')
+
+    async def get_home_adapters(self):
+        '''
+        Retrieve the list of registered home adapters
+        '''
+        return await self._access.get('home/adapters')
+
+    async def get_camera(self):
+        '''
+        Get camera info
+        '''
+        return await self._access.get('camera')
+
+    async def get_camera_snapshot(self, camera_index=0, size=4, quality=5):
+        '''
+        Get camera snapshot
+        '''
+        fbx_cameras = await self.get_camera()
+        return await self._access.get(fbx_cameras[camera_index]['stream_url'].replace('stream.m3u8', f'snapshot.cgi?size={size}&quality={quality}')[1:])
+
+    async def get_camera_stream_m3u8(self, camera_index=0, channel=2):
+        '''
+        Get camera stream
+        '''
+        fbx_cameras = await self.get_camera()
+        return await self._access.get(fbx_cameras[camera_index]['stream_url'].replace('stream.m3u8', f'stream.m3u8?channel={channel}')[1:])
+
+    async def get_camera_ts(self, ts_name, camera_index=0):
+        '''
+        Get camera stream
+        '''
+        fbx_cameras = await self.get_camera()
+        return await self._access.get(fbx_cameras[camera_index]['stream_url'].replace('stream.m3u8', f'{ts_name}')[1:])
+
+    async def get_home_endpoint_value(self, node_id, endpoint_id):
+        '''
+        Get home endpoint value
+        '''
+        return await self._access.get(f'home/endpoints/{node_id}/{endpoint_id}')
+
+    async def get_home_endpoint_values(self, endpoint_list):
+        '''
+        Get home endpoint values
+        '''
+        return await self._access.post('home/endpoints/get', endpoint_list)
+
+    async def set_home_endpoint_value(self, node_id, endpoint_id, home_endpoint_value=home_endpoint_value_schema):
```

Comment:
I don't understand how the schema can be used as a default value: if you call `set_home_endpoint_value` without specifying a `home_endpoint_value`, then `home_endpoint_value['value']` will be equal to `<class 'object'>`, which does not make much sense.

Reply:
This one is tricky because the value can be either a bool, an int, a float or a string, i did not knew how to define it otherwise.

Reply:
The idea i had making it this way was to be able to write code like
....home_endpoint_value_schema['value'] = my_value
....set_home_endpoint_value(1,1)
....set_home_endpoint_value(1,2)
....set_home_endpoint_value(1,3)

Reply:
I don't think changing the global schema would be a good idea, it would mean changing the default values for all later calls, which may be completely unrelated to the current ones. I would rather clone the schema, modify the clone and pass that to the functions:

```python
home_endpoint_value = copy.deepcopy(home_endpoint_value_schema)
home_endpoint_value['value'] = my_value
set_home_endpoint_value(1, 1, home_endpoint_value)
set_home_endpoint_value(1, 2, home_endpoint_value)
```

Reply:
> This one is tricky because the value can be either a bool, an int, a float or a string, i did not knew how to define it otherwise.

I would use `None` instead of `object`.

Reply:
Understood, i'll fix this today.

Reply:
Fixed